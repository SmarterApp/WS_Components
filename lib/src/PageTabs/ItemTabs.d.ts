import * as React from "react";
/**
 * Names of the different tab types.
 * @type {Tabs}
 */
export declare type Tabs = "viewer" | "rubric" | "information";
/**
 * ItemTabsProps props
 * @export
 * @interface ItemTabsProps
 */
export interface ItemTabsProps {
    changedTab: (tab: Tabs) => void;
    selectedTab: Tabs;
    showRubricTab?: boolean;
}
/**
 * ItemTabs displays options for view state of a selected item.
 * @export
 * @class ItemTabs
 * @extends {React.Component<ItemTabsProps, {}>}
 */
export declare class ItemTabs extends React.Component<ItemTabsProps, {}> {
    private select;
    constructor(props: ItemTabsProps);
    /**
     * event handler for view state.
     */
    onClick: (selectedVal: Tabs) => void;
    /**
     * renders individual tab components
     * @param {string} label
     * @param {Tabs} tabType
     * @returns
     */
    renderTab(label: string, tabType: Tabs): JSX.Element;
    render(): JSX.Element;
}
