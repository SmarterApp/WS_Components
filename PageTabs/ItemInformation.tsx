import * as React from 'react';
import * as ItemCardViewModel from '../ItemCardViewModel'
import * as ItemInformationDetail from './ItemInformationDetail';

export interface Props {
    itemCardViewModel: ItemCardViewModel.ItemCardViewModel;
    depthOfKnowledge: string;
    targetDescription: string;
    commonCoreStandardsDescription: string;
    educationalDifficulty: string;
    evidenceStatement: string;
}

export class ItemInformation extends React.Component<Props, {}> {
    renderInformation() {
        return (
            <div className="modal-dialog about-item-modal" role="document">
                <div className="modal-content">
                    <div className="modal-header">
                        <button type="button" className="close" data-dismiss="modal" aria-label="Close">
                            <span aria-hidden="true">&times;</span>
                        </button>
                        <h4 className="modal-title">About This Item</h4>
                    </div>
                    <div className="modal-body">
                        <ItemInformationDetail.ItemInformationDetail 
                            itemCardViewModel={this.props.itemCardViewModel}
                            depthOfKnowledge={this.props.depthOfKnowledge}
                            targetDescription={this.props.targetDescription}
                            commonCoreStandardsDescription={this.props.commonCoreStandardsDescription}
                            educationalDifficulty={this.props.educationalDifficulty}
                            evidenceStatement={this.props.evidenceStatement}
                        />
                    </div>
                    <div className="modal-footer">
                        <button className="btn btn-primary" form="accessibility-form" data-dismiss="modal">Close</button>
                    </div>
                </div>
            </div>
        );
    }

    render() {
        return (
            <div className="modal fade"
                id="about-item-modal-container"
                tabIndex={-1} role="dialog"
                aria-labelledby="About Item Modal"
                aria-hidden="true">
                {this.renderInformation()};
            </div>
        );
    }
}