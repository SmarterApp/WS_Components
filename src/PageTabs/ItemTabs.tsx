import * as React from "react";

export type Tabs = "viewer" | "rubric" | "information";

export interface ItemTabsProps {
  changedTab: (tab: Tabs) => void;
  selectedTab: Tabs;
  showRubricTab?: boolean;
}

export class ItemTabs extends React.Component<ItemTabsProps, {}> {
  private select: Tabs;

  constructor(props: ItemTabsProps) {
    super(props);

    this.state = {
      selectedTab: this.props.selectedTab
    };
  }

  onClick = (selectedVal: Tabs) => {
    this.props.changedTab(selectedVal);
  };

  renderTab(label: string, tabType: Tabs) {
    return (
      <div
        className={
          this.props.selectedTab === tabType
            ? "selected-tab"
            : "tab-not-selected"
        }
        onClick={() => this.onClick(tabType)}
      >
        {label}
      </div>
    );
  }

  render() {
    let rubricTab;
    if (this.props.showRubricTab && this.props.showRubricTab === true) {
      rubricTab = this.renderTab("Rubric and Exemplar", "rubric");
    }

    return (
      <div className="tabs">
        {this.renderTab("Item Viewer", "viewer")}
        {rubricTab}
        {this.renderTab("Item Information", "information")}
      </div>
    );
  }
}
