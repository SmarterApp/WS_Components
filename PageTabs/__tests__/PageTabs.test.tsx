import * as React from 'react';
import * as ReactDOM from 'react-dom';
import { ItemTabs, Tabs } from '../PageTabs';
import { shallow, mount } from 'enzyme';

describe("PageTabs", () => {
    let tabs: Tabs = "viewer";
    const props = {
        changedTab: jest.fn(),
        selectedTab: tabs
    }

    it("matches snapshot", () => {
        expect(shallow(<ItemTabs {...props}/>)).toMatchSnapshot();;
    })
})