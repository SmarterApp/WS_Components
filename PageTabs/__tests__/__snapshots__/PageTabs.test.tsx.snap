// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageTabs matches snapshot 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div
    className="tabs"
>
    <div
        className="selected-tab"
        onClick={[Function]}
    >
        Item Viewer
    </div>
    <div
        className="tab-not-selected"
        onClick={[Function]}
    >
        Rubric and Exemplar
    </div>
    <div
        className="tab-not-selected"
        onClick={[Function]}
    >
        Item Information
    </div>
</div>,
  "nodes": Array [
    <div
      className="tabs"
>
      <div
            className="selected-tab"
            onClick={[Function]}
      >
            Item Viewer
      </div>
      <div
            className="tab-not-selected"
            onClick={[Function]}
      >
            Rubric and Exemplar
      </div>
      <div
            className="tab-not-selected"
            onClick={[Function]}
      >
            Item Information
      </div>
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <ItemTabs
        changedTab={[Function]}
        selectedTab="viewer"
/>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": ItemTabs {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "onClick": [Function],
        "props": Object {
          "changedTab": [Function],
          "selectedTab": "viewer",
        },
        "refs": Object {},
        "state": Object {
          "selectedTab": "viewer",
        },
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div
          className="tabs"
>
          <div
                    className="selected-tab"
                    onClick={[Function]}
          >
                    Item Viewer
          </div>
          <div
                    className="tab-not-selected"
                    onClick={[Function]}
          >
                    Rubric and Exemplar
          </div>
          <div
                    className="tab-not-selected"
                    onClick={[Function]}
          >
                    Item Information
          </div>
</div>,
        "_debugID": 2,
        "_renderedOutput": <div
          className="tabs"
>
          <div
                    className="selected-tab"
                    onClick={[Function]}
          >
                    Item Viewer
          </div>
          <div
                    className="tab-not-selected"
                    onClick={[Function]}
          >
                    Rubric and Exemplar
          </div>
          <div
                    className="tab-not-selected"
                    onClick={[Function]}
          >
                    Item Information
          </div>
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <ItemTabs
    changedTab={[Function]}
    selectedTab="viewer"
/>,
}
`;
